medusa/service/grpc/server.py

```diff
@@ -189,7 +189,7 @@ def BackupStatus(self, request, context):
                     response.finishTime = datetime.fromtimestamp(backup.finished).strftime(TIMESTAMP_FORMAT)
                 else:
                     response.finishTime = ""
-                BackupMan.register_backup(request.backupName, is_async=False)
+                BackupMan.register_backup(request.backupName, is_async=False, overwrite_existing=False)
```

Comment:
**Question:** Why are we registering the backup in `BackupStatus()`? Is there a case where we could check a backup that wasn't registered yet?

Reply:
We need to do a register here because later, at [line 200](https://github.com/thelastpickle/cassandra-medusa/pull/745/files#diff-0af9023e5dba3232ce866a7482e0b302d746bfa87c8040bb6c3516d481767679L200), we call the `record_status_in_response`, which then calls `BackupMan.get_backup_status(backup_name)`. 

If we are a fresh gRPC server (for example a pod that got restarted) and the first thing we do is to answer a backup status request, this would break. It would break because the BackupMan is an in-memory cache of sorts for backup statuses and if we're not the same instance/process that was up when the backup happened, we don't have the in-memory information.

I realise how clunky this is. I barely understand what the BackupMan does, I haven't yet gotten to understand the why. I would not oppose removing it from the code altogether, unless there's a critical function that it does and that I don't yet know about.